### PR TITLE
fix(es_extended/client/main): fix throwables and parachutes

### PR DESCRIPTION
**Pull request by Kenshiin13**
### Description

This PR fixes the handling of throwables and parachutes in the ESX loadout.

---

### Key Changes:
1. **Throwable Ammo Handling**:
   - Throwables were not being tracked properly in ESX. While ESX would set their ammo count to 0, the weapons themselves were not being removed from the loadout, even though the game itself does. This discrepancy caused issues where throwables couldn't be re-added to the loadout by ESX.
   - To fix this, a `throwable` config entry has been added for all throwable weapons in the `es_extended/shared/config/weapons.lua` file. Now, when the ammo count of a throwable reaches 0, the weapon is completely removed from the loadout, allowing ESX to handle them correctly.

2. **Parachute Ammo Handling**:
   - A thread has been introduced to detect when a parachute is used. Once deployed, the parachute’s ammo count is set to 0, and the item is removed from the loadout, similar to throwable weapons.
   - Parachutes are now classified as throwables and will be handled accordingly in the loadout when their ammo count reaches 0.

---

### Affected Weapons:

The following weapons are now properly handled in ESX:

- **WEAPON_GRENADE** (Grenade)
- **WEAPON_BZGAS** (BZ Gas)
- **WEAPON_MOLOTOV** (Molotov Cocktail)
- **WEAPON_STICKYBOMB** (Sticky Bomb)
- **WEAPON_PROXMINE** (Proximity Mines)
- **WEAPON_SNOWBALL** (Snowballs)
- **WEAPON_PIPEBOMB** (Pipe Bombs)
- **WEAPON_BALL** (Baseball) 
- **WEAPON_SMOKEGRENADE** (Smoke Grenade)
- **WEAPON_FLARE** (Flare)
- **WEAPON_ACIDPACKAGE** (Acid Package)
- **GADGET_PARACHUTE** (Parachute)
- **WEAPON_PETROLCAN** (Jerrycan)
- **WEAPON_HAZARDCAN** (Hazardous Jerrycan)

### CODE REVIEW

File: [core]/es_extended/client/main.lua [hunk 1]
```diff
@@ -286,29 +286,48 @@ end
...
+
+    CreateThread(function()
+        local PARACHUTE_OPENING <const> = 1
+        local PARACHUTE_OPEN <const> = 2
+
+        while ESX.PlayerLoaded do
+            local parachuteState = GetPedParachuteState(ESX.PlayerData.ped)
+
+            if parachuteState == PARACHUTE_OPENING or parachuteState == PARACHUTE_OPEN then
+                while GetPedParachuteState(ESX.PlayerData.ped) ~= -1 do Wait(1000) end
```
Comment (mcNuggets1) [inline [core]/es_extended/client/main.lua:326]: Was wenn man in der Zeit ausloggt?

File: [core]/es_extended/client/main.lua [hunk 2]
```diff
@@ -286,29 +286,48 @@ end
...
-                            TriggerServerEvent("esx:updateWeaponAmmo", weapon.name, ammoCount)
-                        end
+            currentWeapon.hash = GetSelectedPedWeapon(ESX.PlayerData.ped)
+
+            if currentWeapon.hash ~= `WEAPON_UNARMED` then
+                local weaponConfig = ESX.GetWeaponFromHash(currentWeapon.hash)
+
+                if weaponConfig then
+                    currentWeapon.ammo = GetAmmoInPedWeapon(ESX.PlayerData.ped, currentWeapon.hash)
+                    while GetSelectedPedWeapon(ESX.PlayerData.ped) == currentWeapon.hash do Wait(500) end
```
Comment (mcNuggets1) [inline [core]/es_extended/client/main.lua:306]: Was wenn man in der Zeit ausloggt?

Comment (mcNuggets1) [reply [core]/es_extended/client/main.lua:306]: Und ist das nicht irrelevant ob man die Waffe gerade wechselt oder nicht?

File: [core]/es_extended/server/classes/player.lua [hunk 3]
```diff
@@ -608,8 +608,17 @@ function CreateExtendedPlayer(playerId, identifier, group, accounts, inventory,
...
+        if not weapon then
+            return
+        end
+
+        weapon.ammo = ammoCount
+
+        if weapon.ammo <= 0 then
+            local index, weaponConfig = ESX.GetWeapon(weaponName)
+            if weaponConfig.throwable then
+                self.removeWeapon(weaponName)
```
Comment (mcNuggets1) [inline [core]/es_extended/server/classes/player.lua:620]: good shit

File: [core]/es_extended/client/main.lua [hunk 4]
```diff
@@ -286,29 +286,48 @@ end
...
-                        if ammoCount ~= currentWeapon.Ammo then
-                            currentWeapon.Ammo = ammoCount
-                            TriggerServerEvent("esx:updateWeaponAmmo", weapon.name, ammoCount)
-                        end
+            currentWeapon.hash = GetSelectedPedWeapon(ESX.PlayerData.ped)
+
+            if currentWeapon.hash ~= `WEAPON_UNARMED` then
+                local weaponConfig = ESX.GetWeaponFromHash(currentWeapon.hash)
+
+                if weaponConfig then
```
Comment (mcNuggets1) [inline [core]/es_extended/client/main.lua:304]: ESX.DoesWeaponExistByHash wäre an der Stelle richtiger aber geht auch so.

**Review by mcNuggets1 — CHANGES_REQUESTED** [review]

File: [core]/es_extended/client/main.lua (hunk 1 above)
Comment (Kenshiin13) [reply [core]/es_extended/client/main.lua:326]: Valid point

File: [core]/es_extended/client/main.lua (hunk 2 above)
Comment (Kenshiin13) [reply [core]/es_extended/client/main.lua:306]: Valid point

**Review by Mycroft-Studios — APPROVED** [review]

**Pull request merged**